### ISSUE TITLE
`valid-jsdoc` false positive: an object pattern in an assignment pattern.

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#6097]
**What version of ESLint are you using?**

master branch.

**What parser (default, Babel-ESLint, etc.) are you using?**

default.

**Please show your full configuration:**

nothing.

**What did you do? Please include the actual source code causing the issue.**

```js
/**
 * @param {string} a - a.
 * @param {object} [obj] - obj.
 * @param {string} obj.b - b.
 * @param {string} obj.c - c.
 * @returns {void}
 */
function foo(a, {b, c} = {}) {
    // empty
}
```

```
> eslint test.js --env es6 --rule valid-jsdoc:2 --no-ignore --no-eslintrc

C:\Users\starc\Documents\GitHub\eslint\test.js
  1:1  error  Expected JSDoc for 'undefined' but found 'obj'  valid-jsdoc

✖ 1 problem (1 error, 0 warnings)

```

**What did you expect to happen?**

No error.

**What actually happened? Please include the actual, raw output from ESLint.**

```
  1:1  error  Expected JSDoc for 'undefined' but found 'obj'  valid-jsdoc
```

**Comment by mysticatea** [comment]
I'm working on this.
